### PR TITLE
[docs-only] Configuration ADR

### PR DESCRIPTION
**Pull request by refs**

### CODE REVIEW

File: docs/ocis/adr/0008-configuration.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
+---
+title: "8. Configuration"
+date: 2021-05-03T15:00:00+01:00
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/ocis/adr
+geekdocFilePath: 0008-configuration.md
+---
+
+# 5. Configuration Management for oCIS
```
Comment (micbar) [inline docs/ocis/adr/0008-configuration.md:9]: pls remove the header 1, it is in the front matter title

Comment (butonic) [reply docs/ocis/adr/0008-configuration.md:9]:
```suggestion
## 5. Configuration Management for oCIS
```

File: docs/ocis/adr/0008-configuration.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+      - [Unsupervised (`ocis proxy`)](#unsupervised-ocis-proxy)
+    + [Other known use cases](#other-known-use-cases)
+    + [Use Cases for Development](#use-cases-for-development)
+    + [Config Loading](#config-loading)
+    + [Start an extension multiple times with different configs (in Supervised mode)](#start-an-extension-multiple-times-with-different-configs-in-supervised-mode)
+    + [Developing Considered Alternatives Further](#developing-considered-alternatives-further)
+    + [Follow up PR's](#follow-up-prs)
+    + [State of the Art](#state-of-the-art)
+
+## Context and Problem Statement
```
Comment (butonic) [inline docs/ocis/adr/0008-configuration.md:41]:
```suggestion
### Context and Problem Statement
```

File: docs/ocis/adr/0008-configuration.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+>0. Command line flag value from user
+>1. Environment variable (if specified)
+>2. Configuration file (if specified)
+>3. Default defined on the flag
+
+An issue arises in point 2, in the sense that configuration file refers to a single file containing the value for the env variable. The CLI framework we use for flag parsing does not support merging config structs with CLI flags. This introduces an inconsistency with the framework: config structs are not supported, and we cannot hook to the lifecycle of the flags parsing to use a file as source and conform to these rules.
+
+Because we solely rely on [structured configuration](https://github.com/owncloud/ocis/blob/master/ocis-pkg/config/config.go) we need a way to modify values in this struct using the provided means urfave/cli gives us (flags, env variables, config files and default value), but since we have different modes of operation (supervised Vs. unsupervised) we have to define a clear line.
+
+## Decision Drivers
```
Comment (butonic) [inline docs/ocis/adr/0008-configuration.md:56]:
```suggestion
### Decision Drivers
```

File: docs/ocis/adr/0008-configuration.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+- Improve experience for the end user.
+- Improve experience for developers.
+- Sane defaults.
+- Sane overrides.
+
+## Considered Options
```
Comment (butonic) [inline docs/ocis/adr/0008-configuration.md:62]:
```suggestion
### Considered Options
```

**Review by micbar — APPROVED** [review]

**Pull request merged**